### PR TITLE
Update DashdotSLO, allow configuration of external reference

### PR DESCRIPTION
**Pull request by janboll**
Update DashdotSLO, and allow it to use an external reference for getting SLO data from Prometheus.

Related schema change: https://github.com/app-sre/qontract-schemas/pull/501

https://issues.redhat.com/browse/APPSRE-8197

### CODE REVIEW

File: reconcile/dashdotdb_slo.py [hunk 1]
```diff
@@ -98,14 +98,19 @@ def _post(self, service_slos: Iterable[ServiceSLO]) -> Optional[Response]:
...
-        for ns in slo_document.namespaces:
-            promurl = ns.cluster.prometheus_url
-            if not ns.cluster.automation_token:
-                LOG.error(
-                    "namespace does not have automation token set %s - skipping", ns
-                )
-                continue
-            promtoken = self._get_automation_token(ns.cluster.automation_token)
+        for namespace_access in slo_document.namespaces:
+            ns = namespace_access.namespace
```
[the remark this answers is not in the document]
Comment (janboll) [reply reconcile/dashdotdb_slo.py:102]: ns is also used below in lines 141, and following.

[the remark this answers is not in the document]
Comment (hemslo) [reply reconcile/dashdotdb_slo.py:102]: oops, what a long method...

**Review by hemslo — APPROVED** [review]
LGTM!

**Review by geoberle — APPROVED** [review]
only nits - LGTM!

File: reconcile/dashdotdb_slo.py [hunk 2]
```diff
@@ -115,7 +120,7 @@ def _get_service_slo(self, slo_document: SLODocumentV1) -> list[ServiceSLO]:
                 prom_response = self._promget(
                     url=promurl,
                     params={"query": (f"{promquery}")},
-                    token=promtoken,
+                    token=promtoken if promtoken else None,
```
[the remark this answers is not in the document]
Comment (janboll) [reply reconcile/dashdotdb_slo.py:123]: No, that's a miskate. removed it

**Pull request merged**